### PR TITLE
Add a couple more examples

### PR DESCRIPTION
**Pull request by mwerezak**
#340

This PR adds a couple of alternative source code examples.

The goal is to add more Pythonic examples, to show how `python-mip` might be integrated into a Python application.

### CODE REVIEW

File: examples/plant_location2.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    mpl.use('Agg')
+    del mpl
+
+import math
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+import mip
+
+if TYPE_CHECKING:
```
Comment (sebheger) [inline examples/plant_location2.py:29]: @mwerezak I never have seen this construct before. From my best knowledge, someone should not use abc things for typing

File: examples/rcpsp2.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+import mip
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator, Collection, Mapping
+
+@dataclass(frozen=True)
+class Resource:
```
Comment (sebheger) [inline examples/rcpsp2.py:14]: CI fails here

File: examples/plant_location2.py [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    from typing import Callable
+    from collections.abc import Sequence
+
+
+@dataclass(frozen=True)
+class CandidateSite:
```
Comment (sebheger) [inline examples/plant_location2.py:35]: CI fails here

**Review by sebheger — CHANGES_REQUESTED** [review]
Please check again. I guess, there is some problems related to the way you want to do type hinting and frozen data classes, but I am not sure

File: examples/plant_location2.py (hunk 1 above)
Comment (mwerezak) [reply examples/plant_location2.py:29]: I see now it's not compatible before Python 3.9.

Comment (mwerezak) [reply examples/plant_location2.py:29]: I see now that compatibility before Python 3.9 depends on the tooling.
https://peps.python.org/pep-0585/#backwards-compatibility

Comment (mwerezak) [reply examples/plant_location2.py:29]: @sebheger Is there a way to run the CI locally?

**Comment by mwerezak** [comment]
Unfortunately I've been unable to follow up on this and I don't know when I will be able to. I'll redo this when that time comes.

**Pull request closed without merging**